proposals/22-04-localize-command.md

```diff
@@ -277,24 +298,34 @@ require this step, and as shown in **Story 2**, users can create kustomization f
   different kustomization files to share configurations. On top of that, if `kustomize build` had the added
   functionality to check for previous downloads of remote references at said global location, `kustomize localize` would
   not need to overwrite the remote references in `target` to the local downloads. As a result, `kustomize localize`
-  would need to neither write to `target` nor copy `target` into `newDir`. The user would not need to create a top-level
-  kustomization file either. <br></br>
+  would need to neither write to `target` nor copy `target` into `newDir`. The user would not need to specify a `scope`
+  either. <br></br>
 
   Despite its advantages, the alternative design violates the self-contained nature of each kustomize layer. Users would
   be unable to upload a fully localized kustomization directory in version control. Furthermore, this alternative
   complicates the existing kustomize workflow by requiring the setup of global environment variables.
+  <br></br>
 
 * The command could, instead of making a copy, modify `target` directly. However, users would not have an easy way to
   undo the command, which is undesirable.
+  <br></br>
+
+* Instead of requiring the user to specify a second argument `scope`, the command could by definition limit its copying
+  to `target`. However, in the case of **Story 1**, the command would force the user to set `target` to `example` in
+  order to include `example/base` in the localization of `example/overlay`. The user would then have to create a
+  kustomization file at `example` that points to `example/overlay` under the `resources` field. The creation of the
+  kustomization file solely for this purpose is messy and more work for the user.
 
 ## Rollout Plan
+
 This command will have at least alpha and GA releases. Depending on user feedback, we may add a beta.
 
 ### Alpha
 
-This release will limit the depth of nested `localized-files` to 1 layer. In other words, the command will ignore remote
-kustomization directory references that originate from within a `localized-files` directory. In addition, this release
-will ignore KRM functions. The command will not output warnings for remote KRM images or exec binaries.
+This release will ignore 
+
+* local kustomization files that `target` references
```

Comment:
Do you mean that in alpha, we will not localize kustomize bases? If yes, I suggest changing the wording to say we will only localize the top-level kustomization file, and not the bases it refers to.

Reply:
I wrote this under the assumption that we want to localize remote bases. I can reword it to "This release will only localize the top-level kustomization file, i.e. `target`, and remote kustomizations that `target` references".

Otherwise if we're only localizing `target`, I'm happy to adopt your wording.

Reply:
> I wrote this under the assumption that we want to localize remote bases.

Yeah that's right! I wasn't sure if you were trying to say that we wouldn't support that in alpha. Since I misunderstood, you can disregard my original comment.

Could you give an example of what you mean "local kustomization files that `target` references"? I'm not sure what this is referring to

Reply:
Have you started the implementation yet? Since Kustomize uses the same code path to process all Kustomizations regardless of source (once the base has been resolved), it might actually be _more_ complicated to get it to stop one layer deep.

Reply:
Change of plan. We'll implement recursion in alpha.